iothub_client/samples/iothub_client_device_cbor_sample/iothub_client_device_cbor_sample.c

```diff
@@ -0,0 +1,956 @@
+    if (rc != IOTHUB_CLIENT_OK)
+    {
+        (void)printf("Failed to set the C2D message callback: return code %d.\n", rc);
+        exit(rc);
+    }
+
+    send_telemetry();
+
+    // Wait for desired property PATCH messages or C2D messages. Send Telemetry.
+    for (uint8_t message_count = 0; message_count < MAX_MESSAGE_COUNT; message_count++)
+    {
+        ThreadAPI_Sleep(TIMEOUT_RECEIVE_MS);
+        send_telemetry();
+    }
+}
+
+static void disconnect_device_client(void)
+{
+    IoTHubDeviceClient_Destroy(iothub_client_handle);
+    IoTHub_Deinit();
+}
+
+static void send_telemetry(void)
+{
+    int rc;
+
+    // Build the telemetry message in CBOR.
+    uint8_t telemetry_payload_buffer[TELEMETRY_MESSAGE_BUFFER_SIZE];
+    size_t telemetry_payload_length;
+    build_cbor_telemetry(
+        telemetry_payload_buffer, sizeof(telemetry_payload_buffer), &telemetry_payload_length);
+    message_handle = IoTHubMessage_CreateFromByteArray(telemetry_payload_buffer, telemetry_payload_length);
+
+    // Set the content type system property value for telemetry messages. This value is
+    // application-defined and must reflect what the service application expects.
+    rc = IoTHubMessage_SetContentTypeSystemProperty(message_handle, CONTENT_TYPE_TELEMETRY);
+    if (rc != IOTHUB_CLIENT_OK)
+    {
+        (void)printf("Failed to set message properties content type for telemetry: return code %d.\n", rc);
+        exit(rc);
+    }
+
+    (void)printf("\ntelemetry message payload: ");
+    for (size_t i = 0; i < telemetry_payload_length; ++i)
+    {
+        (void)printf("%02X ", telemetry_payload_buffer[i]);
+    }
+    (void)printf("\n");
+
+    // Publish the telemetry message. Set callback for telemetry send confirmation.
+    uint64_t id = telemetry_message_id;
+    IoTHubDeviceClient_SendEventAsync(iothub_client_handle, message_handle, send_telemetry_confirm_callback, &id);
+    (void)printf("The device client published the telemetry message. telemetry_message_id: %" PRIu64 ".\n\n", id);
+
+    // The message is copied to the SDK so we can destroy it.
+    IoTHubMessage_Destroy(message_handle);
+    telemetry_message_id++; // Increase for next telemetry message.
+}
+
+//
+// Callbacks
+//
+static void connection_status_callback(IOTHUB_CLIENT_CONNECTION_STATUS result, IOTHUB_CLIENT_CONNECTION_STATUS_REASON reason, void* user_context)
+{
+    (void)reason;
+    (void)user_context;
+
+    if (result == IOTHUB_CLIENT_CONNECTION_AUTHENTICATED)
+    {
+        (void)printf("\nconnection_status_callback: The device client is connected to an IoT Hub.\n\n");
+    }
+    else
+    {
+        (void)printf("\nconnection_status_callback: The device client has been disconnected.\n\n");
+    }
+}
+
+static void get_twin_async_callback(DEVICE_TWIN_UPDATE_STATE update_state, const unsigned char* payload, size_t size, void* userContextCallback)
+{
+    (void)userContextCallback;
+
+    (void)printf("\nget_twin_async_callback payload: ");
```

Comment:
same here :)
```suggestion
    (void)printf("\n-----get_twin_async_callback payload-----");
```

Reply:
totally up to you

Reply:
i'll check it out. thanks for the formatting ideas :)